ReSharper.FSharp/src/Daemon.FSharp/src/FSharpSymbolHighlightingUtil.cs

```diff
@@ -51,22 +54,35 @@ public static string GetMfvHighlightingAttributeId([NotNull] this FSharpMemberOr
 
       var entity = mfv.DeclaringEntity;
       if (mfv.IsModuleValueOrMember && (entity != null && !entity.Value.IsFSharpModule || mfv.IsExtensionMember))
-        return mfv.IsProperty || mfv.IsPropertyGetterMethod || mfv.IsPropertySetterMethod
-          ? FSharpHighlightingAttributeIdsModule.Property
-          : FSharpHighlightingAttributeIdsModule.Method;
+        if (mfv.IsProperty || mfv.IsPropertyGetterMethod || mfv.IsPropertySetterMethod)
+          return mfv.IsExtensionMember
+            ? FSharpHighlightingAttributeIdsModule.ExtensionProperty
+            : FSharpHighlightingAttributeIdsModule.Property;
+        else
+          return mfv.IsExtensionMember
+            ? FSharpHighlightingAttributeIdsModule.ExtensionMethod
+            : FSharpHighlightingAttributeIdsModule.Method;
 
       if (mfv.LiteralValue != null)
         return FSharpHighlightingAttributeIdsModule.Literal;
 
       if (mfv.IsActivePattern)
         return FSharpHighlightingAttributeIdsModule.ActivePatternCase;
 
-      if (mfv.IsMutable || mfv.IsRefCell())
-        return FSharpHighlightingAttributeIdsModule.MutableValue;
-
       if (IsMangledOpName(mfv.LogicalName))
         return FSharpHighlightingAttributeIdsModule.Operator;
 
+      if (mfv.IsValCompiledAsMethod)
```

Comment:
This property is used to distinguish compiled representations of module values, it's not about how F# sees these symbols, I think it should not be used to highlight F# code.
We use it for looking at how F# symbols represented when used from other languages, but never inside F#.

Reply:
Well, my idea was to group methods (like C# methods) and mvf's with IsValCompiledAsMethod together and also have a clear difference between Methods and FSharpFuncs<_>. I thought this bool meant "F# compiler compiles this function as an actual method, not as FSharpFunc<_>. It happens when a function is bound to top level of module or class. Should I recheck this with csharplab or any other IL-viewer?
I had a different highlighting in at first for this kind of methods, but then it seemed kinda misleading and overlapped with a "Method" highlighting.
Do you propose trying to find another way of distinguishing methods and funcs or just group all fsharp funcs together not matter what they compile to?

Reply:
> I thought this bool meant "F# compiler compiles this function as an actual method, not as FSharpFunc<>. It happens when a function is bound to top level of module or class.

Yes, it's true for module values, though, not for classes.

> Do you propose trying to find another way of distinguishing methods and funcs or just group all fsharp funcs together not matter what they compile to?
I think I'd prefer it highlighted based on how it's seen inside F# code, not the compiled representation.

Reply:
JFYI: fixed the wording about `IsValCompiledAsMethod`, it should be more clear now.

Reply:
I see your point. I will remove that check. I guess I have 3 choices now as we won't really show the difference between methods and funcs (everything will be shown as methods).
1. Remove "FSharp Value Function Identifier", use "Method" instead. Remove "FSharp Mutable Value Function" and use "Mutable value". Will lose the ability to separate mutable funcs and mutable values but maybe it's not needed that much?
2. Rename to just "FSharp Function" and "FSharp mutable function" or something to else?
3. Leave them as they are.
Now that we won't make distinction between Methods and Funcs they actually seem kinda redundant.
I will go with whatever decision you make.

Reply:
> we won't really show the difference between methods and funcs (everything will be shown as methods).

It looks it's covered by IsFunctionType check below.

I think just removing this check will make these corner-case-ish values continue to be highlighted as values (which currently seems fine to me), and the symbols that seen as functions should not be affected at all (due to being covered separately).

Reply:
Yeah. But I just want to know what to do with those identifiers I've added. Do we need them at all?

Reply:
Yes, they all seem to be good, thanks!

I'd probably prefer ValueFunction renamed to Function, but it's subjective and is not that important. 🙂

Reply:
Renamed to Function.